### PR TITLE
test(hx-select-control): surf-2025

### PR DESCRIPTION
**Pull request by BY00565233**
## Description

unit test `<hx-select-control>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-2025(unit test case)

## Before you request a review for this PR:

- [x] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [x] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/elements/hx-select-control/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { fixture, expect } from '@open-wc/testing';
+
+/**
+ * <hx-select-control> component tests
+ *
+ * @type HXSelectControlElement
+ *
+ */
+describe('<hx-select-control> component tests', () => {
```
Comment (100stacks) [inline src/elements/hx-select-control/index.spec.js:9]: Please add a test for:

```js
    get controlElement () {
        return this.querySelector('select');
    }
```

Comment (BY00565233) [reply src/elements/hx-select-control/index.spec.js:9]: Changes have been done, Please review

File: src/elements/hx-select-control/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+              >
+              <!-- ... -->
+              </select>
+            <hx-select></hx-select>
+            <label
+              for="selDemo"
+            >
+            Choose an Option
+            </label>
+          </hx-select-control>`;
```
Comment (lalithkarikelli) [inline src/elements/hx-select-control/index.spec.js:24]: Please align the mockup properly (lint indentations).

Comment (BY00565233) [reply src/elements/hx-select-control/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-select-control/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+            expect(shadow).to.be.null;
+        });
+    });
+
+    describe('test getter method', () => {
+        it('markup should contain a #selDemo <select>', async () => {
+            const elSelector = 'hx-select-control > select#selDemo';
+            const id = 'selDemo';
+            const component = /** @type { HXSelectControlElement } */ await fixture(mockup);
```
Comment (100stacks) [inline src/elements/hx-select-control/index.spec.js:59]: Please change variable to `fragment` to indicate it's more complex HTML.

File: src/elements/hx-select-control/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            const query = component.querySelector(elSelector);
```
Comment (100stacks) [inline src/elements/hx-select-control/index.spec.js:60]: Please revise test.  This isn't testing the getter `controlElement`:

```js
get controlElement () {
        return this.querySelector('select');
    }
```

File: src/elements/hx-select-control/index.spec.js (hunk 3 above)
Comment (BY00565233) [reply src/elements/hx-select-control/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-select-control/index.spec.js (hunk 4 above)
Comment (BY00565233) [reply src/elements/hx-select-control/index.spec.js]: Changes have been done, Please review

File: src/elements/hx-select-control/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+        it(`should NOT have a Shadow DOM`, async () => {
+            const component = /** @type {HXSelectControlElement} */ await fixture(template);
+            const shadow = component.shadowRoot;
+
+            expect(shadow).to.be.null;
+        });
+    });
+
+    describe('test Control Element', () => {
```
Comment (100stacks) [inline src/elements/hx-select-control/index.spec.js:55]: Please update your `screenshot` to include this test.

Comment (BY00565233) [reply src/elements/hx-select-control/index.spec.js:55]: Done

**Pull request merged**